src/AppInstallerCLICore/Commands/UpgradeCommand.cpp

```diff
@@ -46,6 +47,7 @@ namespace AppInstaller::CLI
             Argument::ForType(Args::Type::AcceptSourceAgreements),
             Argument::ForType(Execution::Args::Type::CustomHeader),
             Argument{ "all", Argument::NoAlias, Args::Type::All, Resource::String::UpdateAllArgumentDescription, ArgumentType::Flag },
+            Argument{ "include-unknown", Argument::NoAlias, Args::Type::IncludeUnknown, Resource::String::IncludeUnknownArgumentDescription, ArgumentType::Flag }

```

Comment:
@denelon for argument name review.

Reply:
Personally, I find the argument name somewhat awkward for a single app: `winget upgrade SomeApp --include-unknown`
For `upgrade --all` I understand it as "include apps with unknown version", but I don't quite get what we are "including" for a single app.

Reply:
I think if the user explicitly specifies an app, `--include-unknown` would/should be redundant.

I also think `winget upgrade AwesomePackage --all` would be similarly strange.

Reply:
The current behavior is that `upgrade <ID>` doesn’t upgrade a package if already at latest. I had the argument behavior stay the same so the expectation of winget not doing an upgrade unless it has to holds.

Reply:
Community opinion here; I would think that if I specify an ID, it should attempt to be upgraded even if the version is unknown. I would think that even if the version doesn't change, from a user perspective, an upgrade was at least attempted. It could be very confusing if the user knows there is an update, and they try to upgrade it only to be told that no applicable update was found.

Reply:
There's also a message that prints:
```
PS C:\Users\easton> wingetdev upgrade GOG.Galaxy
This package's version number cannot be determined. To upgrade it anyway, add the argument --include-unknown to your previous command.
PS C:\Users\easton>
```
There was some discussion in the issue about whether or not there should be a setting to restore the current behavior (when in doubt, upgrade), but by default I think it should prompt the user to make a decision. The infinite reupgrading is causing more confusion than this is, IMO.

Reply:
Ahh, I must have missed that. I would be fine with that approach, since it is clear and informative